### PR TITLE
fix: summon in Chain Link 1

### PR DESCRIPTION
**Pull request by salix5**
@mercury233 
@purerosefallen 
fix https://github.com/Fluorohydride/ygopro-scripts/pull/2246

Before:
- EVENT_SPSUMMON_SUCCESS of summon in Chain Link 1 and EVENT_CHAIN_END are different timing.
(Break Auxiliary.RegisterMergedDelayedEvent)

After:
- They are the same timing.

### CODE REVIEW

File: operations.cpp [hunk 1]
```diff
@@ -1954,7 +1954,7 @@ int32 field::summon(uint16 step, uint8 sumplayer, card* target, effect* proc, ui
 		process_single_event();
 		raise_event(target, EVENT_SUMMON_SUCCESS, proc, 0, sumplayer, sumplayer, 0);
 		process_instant_event();
-		if(core.current_chain.size() == 0) {
+		if(!action_type) {
```
Comment (mercury233) [inline operations.cpp:1957]: What is `action_type` ? Is it boolean?

**Review by mercury233 — CHANGES_REQUESTED** [review]

File: operations.cpp (hunk 1 above)
Comment (salix5) [reply operations.cpp]: 9022db8dc9192df5e5488e92c0d46f84a8ab867c
action type:
SUMMON_IN_MP (0)
Summon in Main Phase.
It should generate PROCESSOR_POINT_EVENT directly.

SUMMON_IN_CHAIN (1)
Summon in Chain Link.
It should wait EVENT_CHAIN_END.

File: field.h [hunk 2]
```diff
@@ -705,6 +708,10 @@ class field {
 #define GLOBALFLAG_ACTIVATION_COUNT		0x800
 //
 
+//summon action type
+#define SUMMON_IN_MP	0
```
Comment (mercury233) [inline field.h:712]:
```suggestion
#define SUMMON_IN_IDLE	0
```

**Comment by mercury233** [comment]
The additional normal summon of _Fire Formation - Tenki_ can't trigger `EVENT_SUMMON_SUCCESS`

replay:
[summon success.zip](https://github.com/Fluorohydride/ygopro-core/files/14110538/summon.success.zip)

**Comment by salix5** [comment]
confirming

**Pull request merged**